libs/core/i2c.ts

```diff
@@ -16,12 +16,55 @@ namespace pins {
      */
     //% help=pins/i2c-write-number weight=4 group="i2c"
     //% blockId=i2c_writenumber block="i2c write number|at address %address|with value %value|of format %format|repeated %repeated"
-    export function i2cWriteNumber(address: number, value: number, format: NumberFormat, repeated?: boolean): void {
+    export function i2cWriteNumber(address: number, value: number, format?: NumberFormat, repeated?: boolean): void {
+        if (format == undefined)
+            format = NumberFormat.UInt8LE;
         const buf = control.createBuffer(pins.sizeOf(format))
         buf.setNumber(format, 0, value)
         pins.i2cWriteBuffer(address, buf, repeated)
     }
 
+    /**
+     * Writes a value in a I2C register.
+     * @param address I2c address of the device
+     * @param register register index
+     * @param value value to write
+     * @param registerFormat format of the register, default is UInt8BE
+     * @param valueFormat format of the value, default is UInt8LE
+     */
+    //% weight=3 group="i2c"
+    //% blockId=i2c_writereg block="i2c write register|at address $addfress|at register $register|value $value"
+    export function i2cWriteRegister(address: number, register: number, value: number, registerFormat?: NumberFormat, valueFormat?: NumberFormat): number {
+        if (registerFormat === undefined)
```

Comment:
I've only ever seen 8 bit registers. Maybe swap arguments so register format is last? Or nuke it?